core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -1108,6 +1108,18 @@ sealed trait ZIO[-R, +E, +A] extends Serializable { self =>
    */
   final def sandbox: ZIO[R, Cause[E], A] = foldCauseM(ZIO.fail, ZIO.succeed)
 
+  /**
+   * Extracts the optional value, or fails with the given error 'f'.
+   */
+  def someOrFail[B, E1 >: E](f: E1)(implicit ev: A <:< Option[B]): ZIO[R, E1, B] =
+    self.flatMap(ev(_).fold[ZIO[R, E1, B]](ZIO.fail(f))(ZIO.succeed))
+
+  /**
+   * Extracts the optional value, or fails with a [[NoSuchElementException]]
+   */
+  def someOrFailException[B, E1 >: NoSuchElementException](implicit ev: A <:< Option[B], ev2: E <:< E1): ZIO[R, E1, B] =
+    self.map(_.get).mapError(ev2)
```

Comment:
I really don't like the fact that I am using '.get' here and in the other *orFailException functions, since `.get` is unsafe.

The other implementation I tried here was:
```scala
self.mapError(ev2).flatMap(ev(_).fold[ZIO[R, E1, B]](???)(ZIO.succeed))
```
and in the "???" I tried the following:

1. ```scala
self.mapError(ev2)
```
seemed promising, but it returns a ```ZIO[R,E1,A]``` instead of ```ZIO[R,E1,B]``` as needed.
2 . `ZIO.fail` requires an error, And I don't have one(?), just an error transformation function

I Would love some help on how to avoid this.